internal/common/types.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright (C) 2022-2023 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package common
+
+type PodRoleNamePair struct {
+	PodName  string `json:"podName,omitempty"`
+	RoleName string `json:"roleName,omitempty"`
+}
+
+type GlobalRoleSnapshot struct {
```

Comment:
Add more annotation for this struct.

Reply:
done: [a9933fa](https://github.com/apecloud/kubeblocks/pull/5178/commits/a9933fa8619b55f0ddfa4f05a9d7cb5cd8cf3a74)